File: LeetCode/cpp/2390_Removing_Stars_From_a_String.cpp
class Solution {
public:
	string removeStars(string s) {
		string answer;
		stack<char> st;

		for (int i = 0; i < s.length(); i++) {
			if (s[i] != '*') {
				st.push(s[i]);
				continue;
			}

			if (!st.empty()) {
				st.pop();
			}
		}

		while (!st.empty()) {
			answer += st.top();
			st.pop();
		}

		reverse(answer.begin(), answer.end());
		return answer;
	}
};

File: LeetCode/java/2390_Removing_Stars_From_a_String.java
import java.util.Stack;

class Solution {
    public String removeStars(String s) {
        StringBuilder answer = new StringBuilder();
        Stack<Character> stack = new Stack<>();

        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) != '*') {
                stack.push(s.charAt(i));
                continue;
            }

            if (!stack.isEmpty()) {
                stack.pop();
            }
        }

        while (!stack.isEmpty()) {
            answer.append(stack.pop());
        }

        answer.reverse();
        return answer.toString();
    }
}